config.go

```diff
@@ -549,6 +551,63 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
+// getGithubAppAuth returns app id and app private key if it exists.
+// returns false if it does not have both of those fields.
+func (s *Settings) getGithubAppAuth() (int64, []byte, bool) {
+	key, ok := s.Expansions[GithubAppPrivateKey]
+	if s.AuthConfig.Github != nil && s.AuthConfig.Github.AppId != 0 && ok && key != "" {
+		return s.AuthConfig.Github.AppId, []byte(key), true
+	}
+	return 0, nil, false
+}
+
+func (s *Settings) GetGithubAppClient(caller, owner, repo string, retry utility.HTTPRetryFunction, delay utility.HTTPDelayFunction) *http.Client {
+	appId, key, exists := s.getGithubAppAuth()
+	if !exists {
+		return nil
+	}
+	itr, err := ghinstallation.NewAppsTransport(http.DefaultTransport, appId, key)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "failed to create github jwt app transport",
+			"app_id":  appId,
+		}))
+		return nil
+	}
+	jwtClient := utility.GetHTTPClient()
+	jwtClient.Transport = itr
+	defer utility.PutHTTPClient(jwtClient)
+	client := github.NewClient(jwtClient)
+	installationId, _, err := client.Apps.FindRepositoryInstallation(context.Background(), owner, repo)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "failed to find installation id",
+			"owner":   owner,
+			"repo":    repo,
+			"caller":  caller,
+		}))
+		return nil
+	}
+	return getGithubClient(caller, appId, installationId.GetID(), key, retry, delay)
+}
+
+func getGithubClient(caller string, appId, installationId int64, key []byte, retry utility.HTTPRetryFunction, delay utility.HTTPDelayFunction) *http.Client {
+	client := utility.GetHTTPClient()
+	itr, err := ghinstallation.New(client.Transport, appId, installationId, key)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "error creating github client",
+			"caller":  caller,
+		}))
+		return nil
+	}
+	if retry != nil || delay != nil {
+		return utility.GetCustomHTTPRetryableClientWithTransport(itr, retry, delay)
```

Comment:
I believe wrapping a `utility.GetHTTPClient` (above) in another `utility.GetHTTPClient` here will produce a slow HTTP client leak, because even if we `utility.PutHTTPClient` the wrapper, it won't unwrap the underlying client that was originally used. We should only use 1 HTTP client in this code to avoid this.

Reply:
updated to use 1 client